components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -214,26 +213,30 @@ - (void)layoutSubviews {
 }
 
 - (void)centerLayoutAnimated:(BOOL)animated {
+  CGRect contentBoundingRect = UIEdgeInsetsInsetRect(self.bounds, self.contentInsets);
+  CGFloat centerY = CGRectGetMidY(contentBoundingRect);
+  CGFloat centerX = CGRectGetMidX(contentBoundingRect);
+  UIUserInterfaceLayoutDirection layoutDirection = self.mdf_effectiveUserInterfaceLayoutDirection;
+  if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    centerX = CGRectGetWidth(self.bounds) - centerX;
+  }
+
   if (self.titleBelowIcon) {
-    CGPoint iconImageViewCenter =
-        CGPointMake(CGRectGetMidX(self.bounds), CGRectGetHeight(self.iconImageView.bounds) / 2 +
-                    kMDCBottomNavigationItemViewItemInset);
-    BOOL titleVisibilityNever = self.selected &&
-        self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever;
-    BOOL titleVisibilitySelectedNever = !self.selected &&
-        (self.titleVisibility == MDCBottomNavigationBarTitleVisibilitySelected ||
-         self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever);
-    if (titleVisibilityNever) {
-      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
-    } else if (titleVisibilitySelectedNever) {
-      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    BOOL titleHidden =
+        self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever ||
+        (self.titleVisibility == MDCBottomNavigationBarTitleVisibilitySelected && !self.selected);
+    CGFloat iconHeight = CGRectGetHeight(self.iconImageView.bounds);
+    CGFloat labelHeight = CGRectGetHeight(self.label.bounds);
+    CGFloat totalContentHeight = iconHeight;
+    if (!titleHidden) {
+      totalContentHeight += labelHeight + self.contentVerticalMargin;
     }
+    CGPoint iconImageViewCenter =
+        CGPointMake(centerX, centerY - totalContentHeight / 2 + iconHeight / 2);
     CGPoint badgeCenter =
```

Comment:
Should badgeCenter be corrected for RTL? Looks like it's always on the right side, but UITabBar will be swapped for RTL locales.

![uitabbar-rtl](https://user-images.githubusercontent.com/1753199/42050592-63385e92-7ad7-11e8-9468-de1d0b84d2c4.png)
![home-rtl](https://user-images.githubusercontent.com/1753199/42050593-63429b8c-7ad7-11e8-819d-f6719025c8ee.png)

Reply:
Updated to make this rtl